### PR TITLE
fix(npm): resolve dynamic npm imports individually

### PR DESCRIPTION
**Pull request by dsherret**
* https://github.com/denoland/deno_npm/pull/57
* https://github.com/denoland/deno_graph/pull/498

Closes https://github.com/denoland/deno/issues/17802

### CODE REVIEW

File: tests/specs/npm/dynamic_npm_resolution_failure/__test__.jsonc
Comment (dsherret) [inline tests/specs/npm/dynamic_npm_resolution_failure/__test__.jsonc:1]: Main:

```
---- specs::npm::dynamic_npm_resolution_failure ----
command V:\deno\target\debug\deno.exe run -A main.ts
command cwd V:\deno\tests\specs\npm\dynamic_npm_resolution_failure
output path V:\deno\tests\specs\npm\dynamic_npm_resolution_failure\main.out
-- OUTPUT START --
Download http://localhost:4260/chalk
Download http://localhost:4260/@denotest/dep-cannot-parse
error: Error in @denotest/dep-cannot-parse@1.0.0 parsing version requirement for dependency: @denotest/esm-basic@unknown-scheme:unknown

Invalid npm version requirement. Unexpected character.
  unknown-scheme:unknown
  ~

Caused by:
    0: Invalid npm version requirement. Unexpected character.
         unknown-scheme:unknown
         ~
    1: Unexpected character.
         unknown-scheme:unknown
         ~
```

File: tests/specs/npm/dynamic_npm_resolution_failure/main.out
Comment (dsherret) [inline tests/specs/npm/dynamic_npm_resolution_failure/main.out:1]: Main:

```
---- specs::npm::dynamic_npm_resolution_failure ----
command V:\deno\target\debug\deno.exe run -A main.ts
command cwd V:\deno\tests\specs\npm\dynamic_npm_resolution_failure
output path V:\deno\tests\specs\npm\dynamic_npm_resolution_failure\main.out
-- OUTPUT START --
Download http://localhost:4260/chalk
Download http://localhost:4260/@denotest/dep-cannot-parse
error: Error in @denotest/dep-cannot-parse@1.0.0 parsing version requirement for dependency: @denotest/esm-basic@unknown-scheme:unknown

Invalid npm version requirement. Unexpected character.
  unknown-scheme:unknown
  ~

Caused by:
    0: Invalid npm version requirement. Unexpected character.
         unknown-scheme:unknown
         ~
    1: Unexpected character.
         unknown-scheme:unknown
         ~
```

File: cli/resolver.rs [hunk 1]
```diff
@@ -771,9 +763,7 @@ fn resolve_package_json_dep(
 #[derive(Debug)]
 pub struct WorkerCliNpmGraphResolver<'a> {
   npm_resolver: Option<&'a Arc<dyn CliNpmResolver>>,
-  // use a cache per worker so that another worker doesn't clear the
-  // cache of another worker
-  memory_cache: Rc<RefCell<HashMap<String, Arc<NpmPackageInfo>>>>,
```
Comment (dsherret) [inline cli/resolver.rs:776]: After some other refactors, this cache is only a bit of an optimization when many workers are doing npm resolution and the complexity is not worth keeping around for the edge case.

**Review by bartlomieju — APPROVED** [review]
LGTM, this is a really great change 👍

**Pull request merged**